app/src/main/java/com/github/sdp_begreen/begreen/BeGreenApp.kt

```diff
@@ -66,5 +78,75 @@ class BeGreenApp : Application() {
             modules(productionDbModule, geocoderModule)
         }
 
+        // Enable Firebase persistence for offline capabilities
+        Firebase.database.setPersistenceEnabled(true)
+        FirebaseRef.databaseReference.keepSynced(true)
+
+        // Check for connectivity and perform necessary actions
+        val connectedRef = Firebase.database.getReference(".info/connected")
+        connectedRef.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                val connected = snapshot.getValue(Boolean::class.java) ?: false
+                if (connected) {
+                    Log.d("TAG", "connected")
+
+                    // Handle resending of pending posts
+                    val tinyDB = TinyDB(this@BeGreenApp)
+                    val metas = tinyDB.getListObject("metas", TrashPhotoMetadata::class.java)
+                    val users = tinyDB.getListObject("users", User::class.java)
+                    val bitmaps = tinyDB.getListString("bitmaps")
+
+                    // If there are bitmaps, decode them and attempt to update the user
+                    if (bitmaps.isNotEmpty()) {
+                        val b: ByteArray = Base64.decode(bitmaps[0], Base64.DEFAULT)
+                        val bitmap = BitmapFactory.decodeByteArray(b, 0, b.size)
+
+                        // Launch a coroutine to update user with new metadata
+                        applicationScope.launch { updateUser(metas[0] as TrashPhotoMetadata?
+                            , users[0] as User, bitmap
+                        ) }
+
+                        Toast.makeText(this@BeGreenApp, "Resending Pending Posts No."+metas.size, Toast.LENGTH_LONG).show()
+                    }
+
+                } else {
+                    Log.d("TAG", "not connected")
+
+                }
+            }
+
+            override fun onCancelled(error: DatabaseError) {
+                Log.w("TAG", "Listener was cancelled")
+            }
+        })
+    }
+
+    // Function to update user in the database. Takes metadata, user and bitmap as parameters.
+    private suspend fun updateUser(metadata: TrashPhotoMetadata?, user: User, bitmap: Bitmap) {
+
+        // Add new photo to the database
+        val storedMetadata = metadata?.let {
+            db.addTrashPhoto(bitmap, metadata)
+        }
+
+        storedMetadata?.let {
+
+            // Update user's metadata and score based on the new photo
+            user.addPhotoMetadata(metadata)
```

Comment:
For the picture that has not the correct Id in the DB as Ilias correctly mentionned it, you will have to modify this line. Currently you are adding the metadata, but without its ID, as it is generated while storing it in the DB. You not only have to execute the code if `storedMetadata` is not null, but you have to actually use its value to add to the user.

In other world you have to write it like this:

```suggestion
            user.addPhotoMetadata(it)
```

Reply:
Changes made as proposed, thanks!